core/src/test/java/com/linecorp/armeria/server/ServerBuilderTest.java

```diff
@@ -705,4 +705,90 @@ void multipleDomainSocketAddresses() {
                 new ServerPort(DomainSocketAddress.of("/tmp/bar"),
                                SessionProtocol.HTTP, SessionProtocol.HTTPS));
     }
+
+    @Test
+    void baseContextPathApplied() {
+        final Server server = Server
+                .builder()
+                // 1
+                .port(8888, SessionProtocol.HTTP)
```

Comment:
Sorry, Is there some way to test virtualHost with port (e.g. .`virtualHost("*.foo.com:8888")`) without setting a fixed port?

When I set the port as 0 ( to use random port) I've got the following error. 🤔 

```
virtual host port: 8888 (expected: one of [])
java.lang.IllegalStateException: virtual host port: 8888 (expected: one of [])
	at com.google.common.base.Preconditions.checkState(Preconditions.java:715)
	at com.linecorp.armeria.server.ServerBuilder.buildServerConfig(ServerBuilder.java:2056)
	at com.linecorp.armeria.server.ServerBuilder.build(ServerBuilder.java:2002)
	at com.linecorp.armeria.server.ServerBuilderTest.baseContextPathApplied(ServerBuilderTest.java:740)
	at java.lang.reflect.Method.invoke(Method.java:498)
```
So I guessed I should set a fixed port to test it.

I've investigated test codes in this class, there's a similar test to follow.
-> https://github.com/line/armeria/blob/main/core/src/test/java/com/linecorp/armeria/server/ServerBuilderTest.java#L236

Did you mean to set the port using `.http()`.. ?.?

Reply:
I see 😅    If you would like to test virtual hosts with ports, I think the following reference can help:

https://github.com/line/armeria/blob/b604f3f31f4a571da7809e9d524d0b1576c840d4/core/src/test/java/com/linecorp/armeria/server/PortBasedVirtualHostTest.java#L48-L56

Optionally since the test class is convoluted (so it is difficult to use `ServerExtension`), I wouldn't be against just creating a separate test class (either way is fine)

Reply:
thanks! let me handle it.